csvGenerator.ipynb

```diff
@@ -0,0 +1,195 @@
+    "            with open(filename, \"wb\") as file:\n",
+    "                file.write(response.content)\n",
+    "    return path_list  \n",
+    "\n",
+    "#Returns the set of tuples containing, file_ids and file_name for downloading\n",
+    "\n",
+    "\"\"\"\n",
+    "this function returns a set of tuples that are pair of filename and their file ids\n",
+    "set : (tuple(file_id,file_name),...)\n",
+    "files to download in a collection item are located in the following fields:\n",
+    "uploadedFile, extractedText, pages, image, text, ocr, additionalFiles\n",
+    "\"\"\"\n",
+    "\n",
+    "def get_set(items_list:list):\n",
+    "    ids=set()\n",
+    "    for values in items_list:\n",
+    "        #getting the file ids in uploadedFile, extractedText, pages, image, text, ocr, additionalFiles\n",
+    "        ids.add((values[\"uploadedFile\"][\"id\"],values[\"uploadedFile\"][\"filename\"]))\n",
+    "        if values[\"extractedText\"] and values[\"extractedText\"]!= \"None\":\n",
+    "            ids.add((values[\"extractedText\"][\"id\"],values[\"extractedText\"][\"filename\"]))\n",
+    "        if values[\"pages\"] and values[\"pages\"] != \"None\":\n",
+    "            for value in values[\"pages\"]:\n",
+    "                ids.add((value[\"image\"][\"id\"],value[\"image\"][\"filename\"]))\n",
+    "                ids.add((value[\"text\"][\"id\"],value[\"text\"][\"filename\"]))\n",
+    "                ids.add((value[\"ocr\"][\"id\"],value[\"ocr\"][\"filename\"]))\n",
+    "                for file in value[\"additionalFiles\"]:\n",
+    "                    ids.add((file[\"id\"],file[\"filename\"]))\n",
+    "    return ids\n",
+    "   \n",
+    "#Create the CSV file with all the metadata and file path of the downloaded files.\n",
+    "\n",
+    "\"\"\"\n",
+    "this function writes the meta data to the csv file.\n",
+    "uses a flag to write the fileds at the top of the file in the first time.\n",
+    "\"\"\"\n",
+    "\n",
+    "def write_to_csv(csv_name:str,item:list,flag:int):\n",
+    "    with open(csv_name, 'a', newline='') as file:\n",
+    "        writer = csv.writer(file)\n",
+    "        if flag==0:\n",
+    "            fields = list(item.keys())\n",
+    "            writer.writerow(list(fields))\n",
+    "        writer.writerow(list(item.values()))\n",
+    "        \n",
+    "# Add the item to csv_dict that contains all the collection items(with metadata and downloaded file path)\n",
+    "\n",
+    "\"\"\"\n",
+    "adds items to a dictionary object; csv_dict \n",
+    "with the collection items unique key as the key for the dictionary\n",
+    "and the item dictionary as the value for the respective key, \n",
+    "that contains our fields and value/metadata for a\n",
+    "given collection item that we write to the csv file. \n",
+    "It also adds a path property that indicates the location of the downloades files(if any)\n",
+    "\"\"\"\n",
```

Comment:
what's the point of this fucntion? the description should tell me why it is doing what it's doing, what it expects as input, and what it returns.

Reply:
documented the function